client_restaurant/shared/src/commonMain/kotlin/presentation/meals/MealsScreenModel.kt

```diff
@@ -29,18 +29,19 @@ class MealsScreenModel() :
 
     private fun getCuisine() {
         tryToExecute(
-            mangeCousin::getCuisines,
-            ::onGetCuisineSuccess,
+            function = { mangeCousin.getCuisineByResturantId(
+                restaurantId
+            ) },
+            onSuccess = {value->
+                updateState {
+                    it.copy(cuisine = value.toUIState(), selectedCuisine =value.toUIState().first())
+                }
+                getMeals(state.value.selectedCuisine.id)
+            },
```

Comment:
For better teamwork, it's recommended to use private functions instead of lambda functions.